api/app/models/spree/webhooks/subscriber.rb

```diff
@@ -17,6 +18,7 @@ class Subscriber < Spree::Webhooks::Base
       scope :active, -> { where(active: true) }
       scope :inactive, -> { where(active: false) }
 
+      before_validation :generate_secret_key, unless: :secret_key?
```

Comment:
**NOTE:** The only thing I didn't take care of is the webhook subscribers which already exist. Those won't have a secret key as we speak. 

I think it makes sense to add some "helper methods" which can do things such as; regenerating a secret key and/or generating a secret key for all existing subscribers, etc.

I'm not sure about doing it in this PR, but I can do it in a follow-up PR. How has the Spree team handled these kinda migrations in the past?

Reply:
Yes, the usual way to go would be to create a migration that will backfill/generate the data for existing records. 
They get copied into the project when someone upgrades Spree, so this serves this exact purpose